### ISSUE TITLE
[Google Search API] Google AI Overview

### ISSUE THREAD

**Issue by githubfan21** [serpapi/public-roadmap#1622]
As Google now rolled out AI overview to all its users, and many other tools have reports like Semrush. Please start scraping the AI Overview

we need the following 
Ai_status: shortened/expanded/recommended/none  to determine if it is shortened, expanded snippet, or Google recommended ai but no snippet, or doesn’t exist on page
Ai_discription: snippet
Ai_links: links in ai

**Comment by zyc9012** [comment]
It's only available for signed-in accounts for now.

**Comment by aciddjus** [comment]
It also seems to be available only in Chrome.

**Comment by schaferyan** [comment]
Thanks for the issue @githubfan21 . Just in case it wasn't clear, we can't begin scraping this feature until becomes available without being logged into a private account. Since we only scrape public search results pages, we can't scrape anything that can only be accessed by logging in.

As my colleagues mentioned above, AI Overviews only seem to be available when logged into a profile in Chrome, currently.

If you have an example of a search that returns this result type in an incognito browser window, please feel free to share it.

Once this feature is fully public we can begin working on our implementation.

**Comment by NateSkiles** [comment]
A high-volume customer reached out asking for an update on this and saying they have seen AI Overview rolling out for non-logged-in users.

The roll-out seems to be in a limited capacity, as I've only been able to recreate this with the following query: [link](https://www.google.com/search?q=lcd+vs+oled&sca_esv=99e27bbb5f1f1840&sca_upv=1&ei=r0m-ZpW3IOPv0PEPrKntuAo&oq=icd+vs+ol&gs_lp=Egxnd3Mtd2l6LXNlcnAiCWljZCB2cyBvbCoCCAAyChAAGIAEGLEDGAoyBxAAGIAEGAoyBxAAGIAEGAoyBxAAGIAEGAoyBxAAGIAEGAoyBxAAGIAEGAoyBxAAGIAEGAoyBxAAGIAEGAoyBxAAGIAEGAoyBxAAGIAEGApIvbU0UISONFjgqTRwBXgBkAEBmAGcBKABvxGqAQswLjguMC4xLjAuMbgBA8gBAPgBAZgCDqACvBCoAhTCAgoQABiwAxjWBBhHwgIUEAAYgAQYkQIYtAIYigUY6gLYAQHCAh0QABiABBi0AhjUAxjlAhi3AxiKBRjqAhiKA9gBAcICFhAAGAMYtAIY5QIY6gIYjAMYjwHYAQLCAhYQLhgDGLQCGOUCGOoCGIwDGI8B2AECwgIQEAAYgAQYkQIYigUYRhj7AcICCxAAGIAEGJECGIoFwgILEC4YgAQY0QMYxwHCAgUQABiABMICDhAuGIAEGLEDGIMBGNQCwgIREC4YgAQYsQMY0QMYgwEYxwHCAgsQABiABBixAxiDAcICHBAAGIAEGJECGIoFGEYY-wEYlwUYjAUY3QTYAQPCAgoQABiABBhDGIoFwgINEAAYgAQYsQMYQxiKBcICDRAuGIAEGLEDGEMYigXCAhAQLhiABBixAxhDGNQCGIoFwgIQEC4YgAQYsQMYQxiDARiKBcICEBAAGIAEGLEDGEMYgwEYigWYAwmIBgGQBgi6BgQIARgHugYGCAIQARgKugYGCAMQARgTkgcJNS43LjAuMS4xoAf1Sw&sclient=gws-wiz-serp)

![image](https://github.com/user-attachments/assets/1eaeeba9-3ba9-4c1a-8bc9-1c7e5988e255)

While AIO may only be available for a few queries, I'm adding the queued tag to this issue so we can revisit this topic now that it is possible to receive an AIO consistently in an incognito browser.

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000041196?view=List)

**Comment by dethfire** [comment]
Here are several more that can be generated while in incognito

- what is a digital product
- product research
- best thought of the day
- instagram hashtags trending

**Comment by aciddjus** [comment]
Most of the Chrome user agents would successfully return results; non-chrome user agents would not return the AI Overview block.

Here are some examples:

<img width="2560" alt="image" src="https://github.com/user-attachments/assets/827f58b5-8d8d-40ea-acc6-ccf65aed0bd1">

<img width="2560" alt="image" src="https://github.com/user-attachments/assets/bbf26131-36a1-442c-b6f9-8b787c86e00a">

<img width="2560" alt="image" src="https://github.com/user-attachments/assets/bee17968-30d7-4ed2-ac00-cfe50068658d">

It also works for `UK`:

<img width="2560" alt="image" src="https://github.com/user-attachments/assets/6a6c7705-b22b-4ee4-9e3e-f000861069c8">

----------

We are able to extract this data.

**Comment by dethfire** [comment]
I have a report that it can be generated on mobile Safari private mode too

**Comment by alexbarron** [comment]
Received another request:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000069655?view=List)

**Comment by zyc9012** [comment]
While we are implementing this feature, I'd like to collect some ideas about the output JSON structure.

Let's take this AI Overview block as an example (keyword: `drop shipping`):

![drop-shipping](https://github.com/user-attachments/assets/cdefb815-628f-4095-b1a1-ac0352dd7845)

Do you prefer a detailed line by line break down:

```json
{
  "ai_overview": {
    "text_blocks": [
      {
        "type": "paragraph",
        "snippet": "Drop shipping is a retail method where a company sells products online without keeping them in stock. When a customer places an order, the company forwards the order to a supplier, who then ships the product directly to the customer. The company keeps the profits from the sale.",
        "snippet_highlighted_words": [
          "a retail method where a company sells products online without keeping them in stock"
        ],
        "reference_indexes": [
          0,
          2
        ]
      },
      {
        "type": "paragraph",
        "snippet": "Here are some steps to start a drop shipping business:",
        "reference_indexes": [
          4
        ]
      },
      {
        "type": "list",
        "list": [
          {
            "snippet": "Identify a niche: Choose a specific market to target and research competitors.",
            "reference_indexes": [
              4
            ]
          },
          {
            "snippet": "Select a supplier: Find a reliable supplier that has the products you want to sell. You can sign an agreement with a manufacturer, wholesaler, or other retailer.",
            "reference_indexes": [
              2,
              4
            ]
          },
          {
            "snippet": "Create an online store: Design and build an ecommerce website to showcase your products.",
            "reference_indexes": [
              4
            ]
          },
          {
            "snippet": "Market your business: Use social media ads and engaging content to attract customers.",
            "reference_indexes": [
              4
            ]
          }
        ]
      },
      {
        "type": "paragraph",
        "snippet": "Some advantages of drop shipping include:",
        "reference_indexes": [
          0
        ]
      },
      {
        "type": "list",
        "list": [
          {
            "title": "No need for inventory storage",
            "snippet": "You don't need to invest in warehouse space or risk getting stuck with unsold inventory.",
            "reference_indexes": [
              0
            ]
          },
          {
            "title": "More resources for other activities",
            "snippet": "You can allocate more resources to other retail activities such as marketing.",
            "reference_indexes": [
              0
            ]
          }
        ]
      },
      {
        "type": "paragraph",
        "snippet": "However, drop shipping businesses have a high failure rate. Some potential issues include:",
        "reference_indexes": [
          10
        ]
      },
      {
        "type": "list",
        "list": [
          {
            "snippet": "Uncertainty over where the business operates from and where products are manufactured",
            "reference_indexes": [
              10
            ]
          },
          {
            "snippet": "Inconsistent quality of products",
            "reference_indexes": [
              10
            ]
          },
          {
            "snippet": "Dropshippers may only make between 10% and 30% on average, compared to 50% to 70% for traditional e-commerce retailers",
            "reference_indexes": [
              8
            ]
          }
        ]
      }
    ],
    "thumbnail": "https://serpapi.com/searches/66c53c7915afff0a2a48fc6c/images/ada462b58dfff08439da27aa82fde6c082293d76cc6e06be.png",
    "references": [
      {
        "title": "What Is Dropshipping and How Does It Work? (2024) - Shopify",
        "link": "https://www.shopify.com/blog/what-is-dropshipping#:~:text=Dropshipping%20is%20a%20retail%20method,retail%20activities%20such%20as%20marketing.",
        "snippet": "May 22, 2024 — Dropshipping is a retail method where an online store doesn't keep its products in stock. Instead, when a customer mak...",
        "source": "Shopify",
        "index": 0
      },
      {
        "title": "Drop shipping - Wikipedia",
        "link": "https://en.wikipedia.org/wiki/Drop_shipping#:~:text=Amazon%2C%20the%20online%20shopping%20giant,customer%20using%20packaging%20from%20Amazon.",
        "snippet": "Amazon, the online shopping giant, found early success in a dropshipping business model where they could offer over a million diff...",
        "source": "Wikipedia",
        "index": 1
      },
      {
        "title": "Drop-Shipping: What you Need to Know Before You Buy or Sell Online",
        "link": "https://www.michigan.gov/consumerprotection/protect-yourself/consumer-alerts/shopping/before-you-buy-or-sell-online",
        "source": "State of Michigan",
        "index": 2
      },
      {
        "title": "Dropshipping: Does It Actually Work? (Pros + Cons) | BigCommerce",
        "link": "https://www.bigcommerce.com/blog/dropshipping/#:~:text=Is%20dropshipping%20legal?,with%20a%20Dropshipping%20Agreement%20Contract.",
        "snippet": "Is dropshipping legal? Yes, dropshipping is legal. You may run into other legal issues depending on who your supplier is, but drop...",
        "source": "BigCommerce",
        "index": 3
      },
      {
        "title": "How To Start a Dropshipping Business: 9-Step Guide (2024) - Shopify",
        "link": "https://www.shopify.com/blog/how-to-start-dropshipping",
        "snippet": "Jun 28, 2024 — How do I start a dropshipping business? Identify a niche: Select a specific market to target. Research competitors: An...",
        "source": "Shopify",
        "index": 4
      },
      ...
    ]
  }
}
```

Or one `snippet` as a whole?

```json
{
  "ai_overview": {
    "snippet": "Drop shipping is a retail method where a company sells products online without keeping them in stock. When a customer places an order, the company forwards the order to a supplier, who then ships the product directly to the customer. The company keeps the profits from the sale.\nHere are some steps to start a drop shipping business:\nIdentify a niche: Choose a specific market to target and research competitors.\nSelect a supplier: Find a reliable supplier that has the products you want to sell. You can sign an agreement with a manufacturer, wholesaler, or other retailer.\nCreate an online store: Design and build an ecommerce website to showcase your products.\nMarket your business: Use social media ads and engaging content to attract customers.\nSome advantages of drop shipping include:\nYou don't need to invest in warehouse space or risk getting stuck with unsold inventory.\nYou can allocate more resources to other retail activities such as marketing.\nHowever, drop shipping businesses have a high failure rate. Some potential issues include:\nUncertainty over where the business operates from and where products are manufactured\nInconsistent quality of products\nDropshippers may only make between 10% and 30% on average, compared to 50% to 70% for traditional e-commerce retailers",
    "thumbnail": "https://serpapi.com/searches/66c53c7915afff0a2a48fc6c/images/ada462b58dfff08439da27aa82fde6c082293d76cc6e06be.png",
    "references": [
      {
        "title": "What Is Dropshipping and How Does It Work? (2024) - Shopify",
        "link": "https://www.shopify.com/blog/what-is-dropshipping#:~:text=Dropshipping%20is%20a%20retail%20method,retail%20activities%20such%20as%20marketing.",
        "snippet": "May 22, 2024 — Dropshipping is a retail method where an online store doesn't keep its products in stock. Instead, when a customer mak...",
        "source": "Shopify",
        "index": 0
      },
      {
        "title": "Drop shipping - Wikipedia",
        "link": "https://en.wikipedia.org/wiki/Drop_shipping#:~:text=Amazon%2C%20the%20online%20shopping%20giant,customer%20using%20packaging%20from%20Amazon.",
        "snippet": "Amazon, the online shopping giant, found early success in a dropshipping business model where they could offer over a million diff...",
        "source": "Wikipedia",
        "index": 1
      },
      {
        "title": "Drop-Shipping: What you Need to Know Before You Buy or Sell Online",
        "link": "https://www.michigan.gov/consumerprotection/protect-yourself/consumer-alerts/shopping/before-you-buy-or-sell-online",
        "source": "State of Michigan",
        "index": 2
      },
      {
        "title": "Dropshipping: Does It Actually Work? (Pros + Cons) | BigCommerce",
        "link": "https://www.bigcommerce.com/blog/dropshipping/#:~:text=Is%20dropshipping%20legal?,with%20a%20Dropshipping%20Agreement%20Contract.",
        "snippet": "Is dropshipping legal? Yes, dropshipping is legal. You may run into other legal issues depending on who your supplier is, but drop...",
        "source": "BigCommerce",
        "index": 3
      },
      {
        "title": "How To Start a Dropshipping Business: 9-Step Guide (2024) - Shopify",
        "link": "https://www.shopify.com/blog/how-to-start-dropshipping",
        "snippet": "Jun 28, 2024 — How do I start a dropshipping business? Identify a niche: Select a specific market to target. Research competitors: An...",
        "source": "Shopify",
        "index": 4
      },
      ...
    ]
  }
}
```

Feel free to propose your own structure if neither fits your use case.

cc @githubfan21 @dethfire

**Comment by dethfire** [comment]
@zyc9012 thanks for the demo and options. I've reviewed with our engineer and the detailed view is welcome and preferred. Nice work!

**Comment by hartator** [comment]
Thanks @dethfire @zyc9012! 👍

We are still working on refining things internally. We are worried `text_blocks` like this would be a little intense to reconstruct on the user side. Would something like a `rich_snippet` works? 

Like:

```
Drop shipping is a retail method where a company sells products online without keeping them in stock. When a customer places an order, the company forwards the order to a supplier, who then ships the product directly to the customer. The company keeps the profits from the sale.
Here are some steps to start a drop shipping business:
Identify a niche: Choose a specific market to target and research competitors.
Select a supplier: Find a reliable supplier that has the products you want to sell. You can sign an agreement with a manufacturer, wholesaler, or other retailer.
Create an online store: Design and build an ecommerce website to showcase your products.
Market your business: Use social media ads and engaging content to attract customers.
Some advantages of drop shipping include:
You don't need to invest in warehouse space or risk getting stuck with unsold inventory.
You can allocate more resources to other retail activities such as marketing.
However, drop shipping businesses have a high failure rate. Some potential issues include:
Uncertainty over where the business operates from and where products are manufactured
Inconsistent quality of products
Dropshippers may only make between 10% and 30% on average, compared to 50% to 70% for traditional e-commerce retailers
```

Or you need to know what's bolded in the text?

**Comment by alexbarron** [comment]
Another potential customer asking for this. Leaving this here so we can follow up when we release this.

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000070000?view=List)

**Comment by dethfire** [comment]
@hartator the detailed line by line is very useful I would think any average dev would be able to reconstruct for their needs? Having the citation index with the snippet is very useful for advanced analysis.

**Comment by hartator** [comment]
> @hartator the detailed line by line is very useful and I would imagine any average dev would be able to reconstruct for their needs? Having the citation index with the snippet is very useful for advanced analysis.

Got you, thanks again for the feedback! That's helpful in determining a schema here at SerpApi.

**Comment by martin-serpapi** [comment]
Another high-volume customer requested this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000070062?view=List)

**Comment by schaferyan** [comment]
@martin-serpapi and I reached out to users who have expressed interest in this feature in the past:

[Front](https://app.frontapp.com/open/msg_2hln9rz0?key=hvfiLz0Wf-P3wWN-oAv307tQbd_qBxjh)

**Comment by schaferyan** [comment]
Another user provided information about what they are looking for in this feature:

[Front](https://app.frontapp.com/open/msg_2hq9l9ng?key=hvfiLz0Wf-P3wWN-oAv307tQbd_qBxjh)

**Comment by sonika-serpapi** [comment]
> Or you need to know what's bolded in the text?

Ryan and I met with a customer interested in using AI overviews. They mentioned needing bolded text as well. Added notes here: 

[Front](https://app.frontapp.com/inboxes/teammates/24726044/inbox/open/106439528540)

**Comment by zyc9012** [comment]
This feature has been released. Please visit https://serpapi.com/ai-overview for documentation.

![image](https://github.com/user-attachments/assets/6b2d888f-f2f8-4e6b-b35d-1119d42a3780)

**Comment by bigyankarki** [comment]
I understand this is closed, but is it working as expected? We tried querying a few hundred prompts, and all we get is `Can't generate an AI overview right now. Try again later.`. It works in the browser with the same query and for same location.

**Comment by dethfire** [comment]
We've seen the same thing for many keywords we know are generating an AIO. Maybe Google is blocking the serpapi crawler?

**Comment by NateSkiles** [comment]
@bigyankarki From what I'm seeing in our playground ([link](https://serpapi.com/playground?q=drop+shipping&location=Austin%2C+Texas%2C+United+States&gl=us&hl=en&newPara=chips)), this appears to be working as expected:

![image](https://github.com/user-attachments/assets/21c9b8dd-db8f-481b-85ae-65e818042276)

Please reach out to us via chat or email at contact@serpapi.com, and we'd be happy to linvestigate further. It would be helpful if you could send us search IDs where you've observed this behavior as well.

**Comment by NateSkiles** [comment]
@dethfire Would you mind providing some sample queries that generated AI overviews previously but are now receiving this error?

I'll start digging into this in the meantime.

**Comment by bigyankarki** [comment]
@NateSkiles Here is one:

This google search triggers AI overview. [Link](https://www.google.com/search?q=How+does+the+Fast+Food+company+Wendy%27s+excel+on+vegetarian+options+in+United+States%3F&sca_esv=5ac10ce223656a07&sca_upv=1&sxsrf=ADLYWIJ2dHxHTYQyGSarc53YbO4Th80GGA%3A1727299157041&source=hp&ei=VX70Zj6eyZDyD6KegJgO&iflsig=AL9hbdgAAAAAZvSMZRS_GrGDqKiMRjyGE4RSJje_-dNS&ved=0ahUKEwj-_rHlgt-IAxWeJEQIHSIPAOMQ4dUDCBc&uact=5&oq=How+does+the+Fast+Food+company+Wendy%27s+excel+on+vegetarian+options+in+United+States%3F&gs_lp=Egdnd3Mtd2l6IlRIb3cgZG9lcyB0aGUgRmFzdCBGb29kIGNvbXBhbnkgV2VuZHkncyBleGNlbCBvbiB2ZWdldGFyaWFuIG9wdGlvbnMgaW4gVW5pdGVkIFN0YXRlcz9IAFAAWABwAHgAkAEAmAEAoAEAqgEAuAEDyAEA-AEC-AEBmAIAoAIAmAMAkgcAoAcA&sclient=gws-wiz)

While the same search results in `Can't generate an AI overview right now. Try again late`. [Link](https://serpapi.com/playground?q=How+does+the+Fast+Food+company+Wendy%27s+excel+on+vegetarian+options+in+United+States%3F&location=Austin%2C+Texas%2C+United+States&gl=us&hl=en&newPara=chips)

<img width="1427" alt="Screenshot 2024-09-25 at 3 22 03 PM" src="https://github.com/user-attachments/assets/1c85b6f1-7c10-4e5f-b9c6-a116619e3ce7">
<img width="1427" alt="Screenshot 2024-09-25 at 3 22 28 PM" src="https://github.com/user-attachments/assets/ab90fbf1-fc55-4e3a-8025-c31de3c6a86c">

**Comment by NateSkiles** [comment]
@bigyankarki @dethfire I've opened a new issue for this, please feel free to add any context I may have missed: [[Google AI Overview API] No AI Overview generated sporadically #1990](https://github.com/serpapi/public-roadmap/issues/1990).

Thank you for bringing this to our attention!

**Comment by bigyankarki** [comment]
@NateSkiles It seems that before throwing the error, the AI is generating responses. This suggests there might be a chance that Google is intentionally blocking the process.

**Comment by NateSkiles** [comment]
@bigyankarki Thanks for that information! It's quite possible Google could be blocking the process, though as I mentioned [here](https://github.com/serpapi/public-roadmap/issues/1622#issuecomment-2375276529), it doesn't appear all of the AI Overviews are being blocked.

My initial hunch is that Google might be taking longer to generate certain AI Overviews (perhaps they have some cached), and so the content isn't returned to our parser in the initial response.

Either way, we will look into this further and follow up on the other issues with our findings.

**Comment by zyc9012** [comment]
We've implemented a standalone engine when a separate request is needed to fetch AI Overview result. Instead of returning an error, we now return `page_token` and `serpapi_link` for fetching the AI Overview result. For example.

```
  "ai_overview": {
    "page_token": "rWmjgXictZPdkqI4FMcv90nWvWBUtO0e7SpqKiBqHL8ifuENhRADgnwFjHo377JPtU-zAdtunXFqZ6t2U6UkOTkn55xf_t_-9N3A--u3P5w0jehrtcoYq5AwJD6uWOG-atJTYFW3oU8Ty_mSZT6WmKAAiKACTbRanO2unw06Ux93O-dBZ5GZq6m_VuSZ3e2I69WwRHwpoyXHl3BQOh2kp5JFJbGEXSlmm5d1loDu-Ox5BzRxrHY5BGGpuFEyNibFO57QcUerRpX_eb8_tfmiQis4MOZaRYsmSdzw-119WMH0uTKumnt-BPziQB9TRSFIKWYQKHeHyPtpBchAA0QuVj3-U6dFCAigcnHyEDgYTfXO3wJva9nKo80vt0IA2nc5kHzr0ZB5o9vNB4nLBIEuT6RRbF0uQXmQBgTD99JYTwTN9pGMv2rsh9gRIHsw0VbH2zo_Ku8vd65FVKCgqk14X8XqJuGfWjWheZeVWUsMO3F9uNTMZrpqiu3M7kCViaNa92W3mKJPBUGLviOktwihGwf2aD0-U1YZVOTalR7qQagW3VAJICJ51BcFfXROyXMnIPR4fWTA8t6OinJ6-YbFG6Nf3bSrw_KjXD5RYE6PqPBqzvuaD5g3VB0C4HAMQIacnkZ6jTk3tYsAyAO6LrsgBF8BWDPl5ulAcrmjcXlQKsD8hbGhvFKZChB_RQh4edSCHIJyXhAaMvWOXlHoP4z5T_Zl8ABXV1_EZXdAengab2akpZ2yaLJbOGc0vuIK95t_JTnrf0HqckHGb0iVx0hvBMkdVtz-JsiQC_KC1NMLQd4htaD6htS6QQp_GakMGnq7_AgpF-QeyFyQOdJ5gRTp3yP9rwRZkMWXz706sx35HruqlkeeU4u9ujwD09MzEeOTZcv0qCUc-3afvkZJSD4Zrv0q1_vpyS7FUi9kgh1iKqQOFrQoTFLKsJkI_IFEZnASRq6Hha3p-wJ1uFEIA4EG2PRwIriBMA_cFNuClpoppl9Kez8JJdF8YYd4pKvePGUG7O_Ugb8sPwf-4uwaT9vjYVSvRb10arx8Xs2i3uzZH_WV5SrIlL4xaTlKH4VqbwyDhOygbY1OmaqjxdA9U9dtmed2f6e7x3lnvWZav564IjvPxvt6013TcmO8Ao0BMpA8D7F23srrGvDnNivyopLeXWTQ6SxbC1SilmlgepDqm-Zn88ls1uubzab21CwMWXSQan8DmsQweA",
    "serpapi_link": "https://serpapi.com/search.json?engine=google_ai_overview&page_token=rWmjgXictZPdkqI4FMcv90nWvWBUtO0e7SpqKiBqHL8ifuENhRADgnwFjHo377JPtU-zAdtunXFqZ6t2U6UkOTkn55xf_t_-9N3A--u3P5w0jehrtcoYq5AwJD6uWOG-atJTYFW3oU8Ty_mSZT6WmKAAiKACTbRanO2unw06Ux93O-dBZ5GZq6m_VuSZ3e2I69WwRHwpoyXHl3BQOh2kp5JFJbGEXSlmm5d1loDu-Ox5BzRxrHY5BGGpuFEyNibFO57QcUerRpX_eb8_tfmiQis4MOZaRYsmSdzw-119WMH0uTKumnt-BPziQB9TRSFIKWYQKHeHyPtpBchAA0QuVj3-U6dFCAigcnHyEDgYTfXO3wJva9nKo80vt0IA2nc5kHzr0ZB5o9vNB4nLBIEuT6RRbF0uQXmQBgTD99JYTwTN9pGMv2rsh9gRIHsw0VbH2zo_Ku8vd65FVKCgqk14X8XqJuGfWjWheZeVWUsMO3F9uNTMZrpqiu3M7kCViaNa92W3mKJPBUGLviOktwihGwf2aD0-U1YZVOTalR7qQagW3VAJICJ51BcFfXROyXMnIPR4fWTA8t6OinJ6-YbFG6Nf3bSrw_KjXD5RYE6PqPBqzvuaD5g3VB0C4HAMQIacnkZ6jTk3tYsAyAO6LrsgBF8BWDPl5ulAcrmjcXlQKsD8hbGhvFKZChB_RQh4edSCHIJyXhAaMvWOXlHoP4z5T_Zl8ABXV1_EZXdAengab2akpZ2yaLJbOGc0vuIK95t_JTnrf0HqckHGb0iVx0hvBMkdVtz-JsiQC_KC1NMLQd4htaD6htS6QQp_GakMGnq7_AgpF-QeyFyQOdJ5gRTp3yP9rwRZkMWXz706sx35HruqlkeeU4u9ujwD09MzEeOTZcv0qCUc-3afvkZJSD4Zrv0q1_vpyS7FUi9kgh1iKqQOFrQoTFLKsJkI_IFEZnASRq6Hha3p-wJ1uFEIA4EG2PRwIriBMA_cFNuClpoppl9Kez8JJdF8YYd4pKvePGUG7O_Ugb8sPwf-4uwaT9vjYVSvRb10arx8Xs2i3uzZH_WV5SrIlL4xaTlKH4VqbwyDhOygbY1OmaqjxdA9U9dtmed2f6e7x3lnvWZav564IjvPxvt6013TcmO8Ao0BMpA8D7F23srrGvDnNivyopLeXWTQ6SxbC1SilmlgepDqm-Zn88ls1uubzab21CwMWXSQan8DmsQweA"
  }
```

Please see https://serpapi.com/ai-overview#api-examples-example-with-an-extra-request-required for details.

**Comment by marm123** [comment]
@bigyankarki @dethfire tagging for visibility

**Comment by hilmanski** [comment]
Hi @githubfan21 @bigyankarki @dethfire ,

We would love to ask your opinion on returning the AI overview data in a markdown format instead.

So instead of something like:
```
{
  "ai_overview": {
    "text_blocks": [
      {
        "type": "paragraph",
        "snippet": "Drop shipping is a retail method where a company sells products online without keeping them in stock. When a customer places an order, the company forwards the order to a supplier, who then ships the product directly to the customer. The company keeps the profits from the sale.",
        "snippet_highlighted_words": [
          "a retail method where a company sells products online without keeping them in stock"
        ],
        "reference_indexes": [
          0,
          2
        ]
      },
      {
        "type": "paragraph",
        "snippet": "Here are some steps to start a drop shipping business:",
        "reference_indexes": [
          4
        ]
      },
      {
        "type": "list",
        "list": [
          {
            "snippet": "Identify a niche: Choose a specific market to target and research competitors.",
            "reference_indexes": [
              4
            ]
          },
          {
            "snippet": "Select a supplier: Find a reliable supplier that has the products you want to sell. You can sign an agreement with a manufacturer, wholesaler, or other retailer.",
            "reference_indexes": [
              2,
              4
            ]
          },
          {
            "snippet": "Create an online store: Design and build an ecommerce website to showcase your products.",
            "reference_indexes": [
              4
            ]
          },
          {
            "snippet": "Market your business: Use social media ads and engaging content to attract customers.",
            "reference_indexes": [
              4
            ]
          }
        ]
      },
      {
        "type": "paragraph",
        "snippet": "Some advantages of drop shipping include:",
        "reference_indexes": [
          0
        ]
      },
      {
        "type": "list",
        "list": [
          {
            "title": "No need for inventory storage",
            "snippet": "You don't need to invest in warehouse space or risk getting stuck with unsold inventory.",
            "reference_indexes": [
              0
            ]
          },
          {
            "title": "More resources for other activities",
            "snippet": "You can allocate more resources to other retail activities such as marketing.",
            "reference_indexes": [
              0
            ]
          }
        ]
      },
      {
        "type": "paragraph",
        "snippet": "However, drop shipping businesses have a high failure rate. Some potential issues include:",
        "reference_indexes": [
          10
        ]
      },
      {
        "type": "list",
        "list": [
          {
            "snippet": "Uncertainty over where the business operates from and where products are manufactured",
            "reference_indexes": [
              10
            ]
          },
          {
            "snippet": "Inconsistent quality of products",
            "reference_indexes": [
              10
            ]
          },
          {
            "snippet": "Dropshippers may only make between 10% and 30% on average, compared to 50% to 70% for traditional e-commerce retailers",
            "reference_indexes": [
              8
            ]
          }
        ]
      }
    ],
    "thumbnail": "https://serpapi.com/searches/66c53c7915afff0a2a48fc6c/images/ada462b58dfff08439da27aa82fde6c082293d76cc6e06be.png",
    "references": [
      {
        "title": "What Is Dropshipping and How Does It Work? (2024) - Shopify",
        "link": "https://www.shopify.com/blog/what-is-dropshipping#:~:text=Dropshipping%20is%20a%20retail%20method,retail%20activities%20such%20as%20marketing.",
        "snippet": "May 22, 2024 — Dropshipping is a retail method where an online store doesn't keep its products in stock. Instead, when a customer mak...",
        "source": "Shopify",
        "index": 0
      },
      }
      ...
```

We'll return this markdown instead for the text
```
{
  "ai_overview": {
    "text": "Drop shipping is a retail method where a company sells products online without keeping them in stock. When a customer places an order, the company forwards the order to a supplier, who then ships the product directly to the customer. The company keeps the profits from the sale.\n\nHere are some steps to start a drop shipping business:\n\n- Identify a niche: Choose a specific market to target and research competitors.\n- Select a supplier: Find a reliable supplier that has the products you want to sell. You can sign an agreement with a manufacturer, wholesaler, or other retailer.\n- Create an online store: Design and build an ecommerce website to showcase your products.\n- Market your business: Use social media ads and engaging content to attract customers.\n\nSome advantages of drop shipping include:\n\n**No need for inventory storage**\nYou don't need to invest in warehouse space or risk getting stuck with unsold inventory.\n\n**More resources for other activities**\nYou can allocate more resources to other retail activities such as marketing.\n\nHowever, drop shipping businesses have a high failure rate. Some potential issues include:\n- Uncertainty over where the business operates from and where products are manufactured\n- Inconsistent quality of products\n- Dropshippers may only make between 10% and 30% on average, compared to 50% to 70% for traditional e-commerce retailers",
    "thumbnail": "https://serpapi.com/searches/66c53c7915afff0a2a48fc6c/images/ada462b58dfff08439da27aa82fde6c082293d76cc6e06be.png",
    "references": [
      {
        "title": "What Is Dropshipping and How Does It Work? (2024) - Shopify",
        "link": "https://www.shopify.com/blog/what-is-dropshipping#:~:text=Dropshipping%20is%20a%20retail%20method,retail%20activities%20such%20as%20marketing.",
        "snippet": "May 22, 2024 — Dropshipping is a retail method where an online store doesn't keep its products in stock. Instead, when a customer mak...",
        "source": "Shopify",
        "index": 0
      },
      ....
```

Thank you very much for your help!
